ext/openssl/ossl.c

```diff
@@ -1111,10 +1111,14 @@ Init_openssl(void)
      */
     /* CRYPTO_malloc_init(); */
     /* ENGINE_load_builtin_engines(); */
+#if !defined(LIBRESSL_VERSION_NUMBER) && (OPENSSL_VERSION_NUMBER >= 0x10100000L)
+    OPENSSL_init_ssl(0, NULL);
+#else
     OpenSSL_add_ssl_algorithms();
     OpenSSL_add_all_algorithms();
     ERR_load_crypto_strings();
     SSL_load_error_strings();
+#endif
```

Comment:
Suffix `L` and the parentheses around `OPENSSL_VERSION_NUMBER >= ~~` are redundant.

The return value from OPENSSL_init_ssl() should be checked.

    if (!OPENSSL_init_ssl(0, NULL))
        rb_raise(rb_eRuntimeError, "OPENSSL_init_ssl");

While you're at it, can you remove the obsolete comments above and below?